tests_pytest/pytorch_tests/e2e_tests/post_training_quantization_tests/test_quantization_error_methods.py

```diff
@@ -0,0 +1,143 @@
+def rep_data_gen():
+    """
+    Fixture to create a representative dataset generator for post-training quantization.
+
+    Generates a small dataset based on the defined INPUT_SHAPE.
+    """
+    np.random.seed(42)
+
+    def representative_dataset():
+        for _ in range(2):
+            yield [np.random.randn(*INPUT_SHAPE)]
+
+    return representative_dataset
+
+
+def get_tpc():
+    """
+    Create a target platform capabilities (TPC) configuration with no weight quantization.
+
+    Returns a TPC object for quantization tests.
+    """
+    att_cfg_noquant = AttributeQuantizationConfig()
+
+    op_cfg = OpQuantizationConfig(default_weight_attr_config=att_cfg_noquant,
+                                  attr_weights_configs_mapping={KERNEL_ATTR: att_cfg_noquant,
+                                                                BIAS_ATTR: att_cfg_noquant},
+                                  activation_quantization_method=QuantizationMethod.UNIFORM,
+                                  activation_n_bits=2,
+                                  supported_input_activation_n_bits=2,
+                                  enable_activation_quantization=True,
+                                  quantization_preserving=True,
+                                  fixed_scale=None,
+                                  fixed_zero_point=None,
+                                  simd_size=32,
+                                  signedness=Signedness.AUTO)
+
+    tpc = generate_tpc(default_config=op_cfg, base_config=op_cfg, mixed_precision_cfg_list=[op_cfg], name="test_tpc")
+
+    return tpc
+
+
+def get_core_config(quant_error_method):
+    """
+    Create a core configuration with a specified quantization error method.
+
+    Parameters:
+        quant_error_method: QuantizationErrorMethod to be used in the configuration.
+
+    Returns:
+        CoreConfig instance configured with the specified quantization error method.
+    """
+    quantization_config = QuantizationConfig(activation_error_method=quant_error_method)
+    return CoreConfig(quantization_config=quantization_config)
+
+
+class TestPTQWithQuantizationErrorMethods:
+    def test_ptq_quantization_error_methods(self, rep_data_gen):
+        """
+        Verify that post-training quantization produces different quantization parameters based
+        on the chosen quantization error method.
+
+        For each quantization error method (MSE, HMSE, MAE, NOCLIPPING), this test builds a model,
+        applies post-training quantization, and records the max_range of the ReLU activation quantizer.
+        It then asserts that each method produces a distinct quantization parameter.
+        """
+        model = build_model(input_shape=INPUT_SHAPE, out_chan=16, kernel=1, const=np.array([5]))
+        tpc = get_tpc()
+        quant_max_range={}
+        quant_error_method_types = [QuantizationErrorMethod.MSE, QuantizationErrorMethod.HMSE, QuantizationErrorMethod.MAE, QuantizationErrorMethod.NOCLIPPING]
```

Comment:
can be a test parameter, can't it? (that way, a separate test instance would run for each error method)